### PR TITLE
Added default values for CA flags

### PR DESCRIPTION
**Pull request by aaronfern**
**How to categorize this PR?**

/area documentation open-source
/kind technical-debt

**What this PR does / why we need it**:
This PR adds default values for the following two `cluster-autoscaler` flags

1. max-empty-bulk-delete
2. new-pod-scale-up-delay

This ensures that g/g always controls the default values passed to these flags, and users can quickly find out default values from the g/g codebase itself

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
/cc @himanshu-kun 

**Release note**:

```other operator
None
```

### CODE REVIEW

File: pkg/apis/core/v1beta1/defaults_shoot.go [hunk 1]
```diff
@@ -329,6 +329,12 @@ func SetDefaults_ClusterAutoscaler(obj *ClusterAutoscaler) {
 	if obj.MaxGracefulTerminationSeconds == nil {
 		obj.MaxGracefulTerminationSeconds = pointer.Int32(600)
 	}
+	if obj.NewPodScaleUpDelay == nil {
+		obj.NewPodScaleUpDelay = &metav1.Duration{Duration: 0}
+	}
+	if obj.MaxEmptyBulkDelete == nil {
+		obj.MaxEmptyBulkDelete = pointer.Int32(10)
+	}
```
Comment (ary1992) [inline pkg/apis/core/v1beta1/defaults_shoot.go:343]: Adapt unit tests as well.

**Review by ary1992 — COMMENTED** [review]
Mention the default values in the `pkg/apis/core/v1beta1/types_shoot.go` as well
https://github.com/gardener/gardener/blob/05d05ff710e7e0dde7630425e6929ef87de58bb0/pkg/apis/core/v1beta1/types_shoot.go#L601

**Review by himanshu-kun — COMMENTED** [review]
/lgtm overall changes wise... pls address the changes requested by g/g maintainers regarding specifics

File: pkg/apis/core/v1beta1/defaults_shoot.go (hunk 1 above)
Comment (aaronfern) [reply pkg/apis/core/v1beta1/defaults_shoot.go:343]: Done

File: pkg/apis/core/v1beta1/defaults_shoot_test.go [hunk 2]
```diff
@@ -896,6 +898,8 @@ var _ = Describe("Shoot defaulting", func() {
 				MaxGracefulTerminationSeconds: pointer.Int32(60 * 60 * 24),
 				IgnoreDaemonsetsUtilization:   pointer.Bool(true),
 				Verbosity:                     pointer.Int32(4),
+				NewPodScaleUpDelay:            &metav1.Duration{Duration: 7 * time.Hour},
+				MaxEmptyBulkDelete:            pointer.Int32(60 * 60 * 24),
```
Comment (ary1992) [inline pkg/apis/core/v1beta1/defaults_shoot_test.go]:
```suggestion
				NewPodScaleUpDelay:            &metav1.Duration{Duration: 1},
				MaxEmptyBulkDelete:            pointer.Int32(20),
```

**Review by ary1992 — COMMENTED** [review]
Minor nit.

**Comment by ary1992** [comment]
Thanks for addressing the feedbacks.
/lgtm
/approve

File: pkg/component/clusterautoscaler/cluster_autoscaler_test.go [hunk 3]
```diff
@@ -89,6 +89,8 @@ var _ = Describe("ClusterAutoscaler", func() {
 		configScanInterval                        = &metav1.Duration{Duration: time.Second}
 		configIgnoreDaemonsetsUtilization   bool  = true
 		configVerbosity                     int32 = 4
+		configMaxEmptyBulkDelete                  = pointer.Int32(10)
```
Comment (shafeeqes) [inline pkg/component/clusterautoscaler/cluster_autoscaler_test.go]: Maybe use a different value here rather than the default?

Comment (aaronfern) [reply pkg/component/clusterautoscaler/cluster_autoscaler_test.go]: Done

**Pull request merged**